src/cli/compiler.js

```diff
@@ -117,6 +117,7 @@ class Compiler {
           chalk.grey('File change detected: '),
           compiler.displayPath(path)
         )
+        compiler.logger.currentStep = 0
```

Comment:
Can we please do this in a place where it is _guaranteed_ to be reset whenever we compile — i.e. at the beginning of the `compile()` function?